device_backends/include/NumericAddressedBackend.h

```diff
@@ -42,6 +42,20 @@ namespace ChimeraTK {
      */
     virtual bool canMergeRequests() const { return true; }
 
+    /**
+     * @brief Determines the supported minimum alignment for any read/write requests.
+     *
+     * If the backend expects a particular alignment for read()/write() calls it should return a value biggern than 1.
+     * The address and sizeInBytes arguments of the read()/write() calls will be always an integer multiple of this
+     * number. Any unaligned transfers will be changed to meet these criteria (additional padding data will be thrown
+     * away).
+     *
+     * The default implementation returns 1, which means no special alignment is required.
+     *
+     * @return Minimum alignment in bytes
+     */
+    virtual size_t minimumTransferAlignment() const { return 1; }
```

Comment:
This is not true. So far it was implicitly 1, since the NumericAddressedBackend did not perform any alignment. Only the PCIe backend assumed an alignment of 4. We could have implemented support for unaligned operations into all backends which require a particular alignment, and the framework would have been fine. This added feature is merely taking away the need for such implementations from the backends.

In any case, this cannot do any harm: If we would not override this function in the PCIe backend, the behaviour would be identical to before the change: The backend throws when an unaligned operation is attempted.

Reply:
ok